### DISCUSSION TITLE
how can i change the parameterized value when excute the testcase

### DISCUSSION THREAD

**Discussion by sinaparker** [pytest-dev/pytest#8063] [Q&A]
hi，I have a question like below
i have set a global varible ,but i have change it according excute the test case and return a new value
how can i use the new value for the below cases,because pytest collected the values before it excute the cases
code like

```python
import pytest

a="aaa"

class Test_1():

    def test_000(self):
        global a
        a="xxx"

    @pytest.mark.parametrize("stringinput",[a])
    def test_valid_string(self,stringinput):
        assert stringinput.isdigit()

if __name__ == '__main__':
    pytest.main(["-sq","-v","--setup-show",__file__])
```

the output:
```python-traceback
================================== FAILURES ===================================
________________________ Test_1.test_valid_string[aaa] ________________________

self = <test000.Test_1 object at 0x000001C555C56FD0>, stringinput = 'aaa'

    @pytest.mark.parametrize("stringinput",[a])
    def test_valid_string(self,stringinput):
>       assert stringinput.isdigit()
E       AssertionError: assert False
E        +  where False = <built-in method isdigit of str object at 0x000001C55467C148>()
E        +    where <built-in method isdigit of str object at 0x000001C55467C148> = 'aaa'.isdigit
```

the parameterized value is not xxx but aaa 

is there a method i can use the updated value like xxx ? ths for your reply!!!

**Comment by The-Compiler** [comment]
I took the freedom to convert this to a discussion, since this is a question rather than an issue in pytest.

I don't think there's a way to do this at the moment - like you say, collection happens before execution of the tests. What problem are you trying to solve exactly?

**Reply by RonnyPfannschmidt** [reply]
@sinaparker  please outline **what** you want to do instead of how you want to do it,

the content of the question indicates you are very new to python programming 
and the solution may look very different from what you think it would look like

**Reply by sinaparker** [reply]
> I took the freedom to convert this to a discussion, since this is a question rather than an issue in pytest.
> 
> I don't think there's a way to do this at the moment - like you say, collection happens before execution of the tests. What problem are you trying to solve exactly?

maybe it's a limit of pytest frame, i would like to find some other resolution,ths a lot
